src/JSInterop/Microsoft.JSInterop/src/DotNetDispatcher.cs

```diff
@@ -263,16 +251,24 @@ static bool IsIncorrectDotNetObjectRefUse(JsonElement item, Type parameterType)
         /// </exception>
         public static void EndInvoke(string arguments)
         {
-            var parsedArgs = ParseArguments(
-                nameof(EndInvoke),
-                arguments,
-                EndInvokeParameterTypes);
+            var jsRuntimeBase = (JSRuntimeBase)JSRuntime.Current;
+            var utf8JsonBytes = Encoding.UTF8.GetBytes(arguments);
+            var reader = new Utf8JsonReader(utf8JsonBytes);
 
-            EndInvoke((long)parsedArgs[0], (bool)parsedArgs[1], (JSAsyncCallResult)parsedArgs[2]);
-        }
+            reader.Read();
+            if (reader.TokenType != JsonTokenType.StartArray)
+            {
+                throw new JsonException("Invalid JSON");
+            }
+
+            reader.Read();
+            var taskId = reader.GetInt64();
 
-        private static void EndInvoke(long asyncHandle, bool succeeded, JSAsyncCallResult result)
-            => ((JSRuntimeBase)JSRuntime.Current).EndInvokeJS(asyncHandle, succeeded, result);
+            reader.Read();
+            var success = reader.GetBoolean();
+
+            jsRuntimeBase.EndInvokeJS(taskId, success, ref reader);
```

Comment:
Maybe read once more before passing the refer to `EndInvokeJS`? Looks like both branches within `EndInvokeJS` do a `.Read()` first anyway which we can remove.